github/git_trees.go

```diff
@@ -30,6 +30,13 @@ func (t Tree) String() string {
 // TreeEntry represents the contents of a tree structure. TreeEntry can
 // represent either a blob, a commit (in the case of a submodule), or another
 // tree.
+//
+// When used with [GitService.CreateTree], set Content for small text files,
+// or set SHA to reference an existing blob (use [GitService.CreateBlob] for
+// binary files or large content). To delete an entry, set both Content and SHA
```

Comment:
```suggestion
// or set `SHA` to reference an existing blob (use [GitService.CreateBlob] for
// binary files or large content). To delete an entry, set both `Content` and `SHA`
```

Reply:
That wouldn't fit with the [doc links syntax](https://go.dev/doc/comment#doclinks). I guess we could use `[TreeEntry.Content]` and `[TreeEntry.SHA]` instead, but given that other doc comments in this package already refer to fields without such a qualifier, that seems a bit unnecessary to me. Happy to change to do it anyway if you think it's worthwhile.